tidy3d/components/scene.py

```diff
@@ -246,6 +260,36 @@ def medium_map(self) -> Dict[StructureMediumType, pd.NonNegativeInt]:
 
         return {medium: index for index, medium in enumerate(self.mediums)}
 
+    @cached_property
+    def sorted_structures(self) -> List[Structure]:
+        """Returns a list of sorted structures based on the priority strategy when structures
+        overlap. In the sorted list, latter added structures take higher priority.
+
+        Returns
+        -------
+        List[:class:`.Structure`]
+        """
+
+        if self.overlapping_priority == "latter":
+            return self.structures
+
+        # The other strategy is "metal":
+        #   Prioritize metallic structures including LossyMetal and PEC. When two metallic structures overlap,
+        #   prioritize the structure of higher conductivity.
+        def structure_comparator(struct1, struct2):
+            mat_list = [struct1.medium, struct2.medium]
+            # For implementation convenience, let's denote conductivity of regular medium as -1,
+            # so that it's lower than that of any metal.
+            sigmas = [-1, -1]
+            for ind, mat in enumerate(mat_list):
+                if mat.is_pec:
+                    sigmas[ind] = LARGE_NUMBER
+                elif isinstance(mat, LossyMetalMedium):
+                    sigmas[ind] = mat.conductivity
+            return sigmas[0] - sigmas[1]
+
+        return sorted(self.structures, key=cmp_to_key(structure_comparator))
```

Comment:
There was some rearrangement done for 2D media too, right? I think it was to put the equivalent volumetric structures on top. I wonder how this plays with these changes, and if it's at all possible to unify these rearrangements more - also to clarify even better to the user what exactly happens in various cases (metals + 2D media? very strange case I guess but still)

Reply:
In this PR and also previously, 2D media is not handled differently from 3D medium. They stay the same order / priority.

Reply:
Only PEC is reordered? Is this clear to the user?

Reply:
if one sets `structure_priority_mode='conductor'`, 2D PEC is reordered. That is expected behavior for RF users.

Reply:
Ah right. Makes sense then